app/lib/features/space/settings/pages/visibility_accessibility_page.dart

```diff
@@ -0,0 +1,310 @@
+        ),
+        borderRadius: BorderRadius.circular(6),
+      ),
+      child: ListTile(
+        title: Text(
+          spaceItem.spaceProfileData.displayName ?? spaceItem.roomId,
+        ),
+        leading: avatar,
+        trailing: IconButton(
+          onPressed: () => removeSpace(space, spaceItem),
+          icon: Icon(
+            Atlas.trash,
+            color: Theme.of(context).colorScheme.error,
+          ),
+        ),
+      ),
+    );
+  }
+
+  Future<void> removeSpace(Space? space, SpaceItem spaceItem) async {
+    final room =
+        await ref.read(maybeRoomProvider(space!.getRoomIdStr()).future);
+    if (room != null) {
+      if (await space.isChildSpaceOf(spaceItem.roomId)) {
+        await room.removeParentRoom(spaceItem.roomId, null);
+      } else {
+        await space.removeChildRoom(spaceItem.roomId, null);
+      }
+    }
+    setState(() {
+      spaceList.remove(spaceItem);
+      if (spaceList.isEmpty) {
+        selectSpace(space.getRoomIdStr());
+      }
+    });
+  }
+
+  Future<void> selectSpace(String roomId) async {
+    try {
+      final spaceId = await selectSpaceDrawer(context: context);
+      if (spaceId != null) {
+        final spaceItem =
+            await ref.watch(maybeSpaceInfoProvider(spaceId).future);
+        final isAlreadyAdded =
+            spaceList.any((element) => element.roomId == spaceItem?.roomId);
+        if (spaceItem != null && !isAlreadyAdded) {
+          spaceList.add(spaceItem);
+          updateSpaceVisibility(
+            RoomVisibility.SpaceVisible,
+            spaceId,
+            roomId: roomId,
+          );
+          setState(() {});
+        }
+      }
+    } catch (e, st) {
+      _log.severe('Select Space Error =>>', e, st);
+      if (!mounted) return;
+      EasyLoading.showToast(
+        L10n.of(context).failedToLoadSpace(e),
+        toastPosition: EasyLoadingToastPosition.bottom,
+      );
+    }
+  }
+
+  Future<void> updateSpaceVisibility(
+    RoomVisibility value,
+    String spaceId, {
+    String? roomId,
+  }) async {
+    try {
+      EasyLoading.show(status: 'Updating space settings', dismissOnTap: false);
+      final sdk = await ref.read(sdkProvider.future);
+      final room = await ref.read(maybeRoomProvider(spaceId).future);
+      final space = await ref.read(maybeSpaceProvider(spaceId).future);
+      final update = sdk.api.newJoinRuleBuilder();
+      switch (value) {
+        case RoomVisibility.Public:
+          update.joinRule('public');
+          break;
+        case RoomVisibility.Private:
+          update.joinRule('invite');
+          break;
+        case RoomVisibility.SpaceVisible:
+          update.joinRule('restricted');
+          if (roomId != null) {
+            if (!(await space!.isChildSpaceOf(roomId))) {
+              await room!.addParentRoom(roomId, true);
+              break;
+            } else {
+              await space.addChildRoom(roomId);
+              break;
+            }
```

Comment:
As per comment above. When changing the list of spaces for the join rule we must provide the entire new list (of each room) through the update builder. And we do not want to touch the parent-child-relationship, as we might not even have the permissions to do so.

so, assuming we are passed a `newSpaceIds`, which has already been properly updated prior to this function being called, we should instead:
```suggestion
            for (final roomId in (newSpaceIds ?? [])) {
              update.addRoom(roomId);
            }
```

And if we now also did the same for removing: just took the room out of that list and submitted it rh